tfs/src/main/java/hudson/plugins/tfs/model/AbstractHookEvent.java

```diff
@@ -294,15 +313,24 @@ private Boolean branchMatches(final GitCodePushedEventArgs gitCodePushedEventArg
                     for (final RemoteConfig repository : git.getRepositories()) {
                         boolean repositoryMatches = false;
                         for (final URIish remoteURL : repository.getURIs()) {
-                            if (UriHelper.areSameGitRepo(uri, remoteURL)) {
+                            String url = this.envVars.expand(remoteURL.toString());
+                            URIish urlWithExpandedParams = null;
+                            try {
+                                urlWithExpandedParams = new URIish(url);
+                            } catch (URISyntaxException e) {
+                                LOGGER.warning(String.format("Remote URL '%s' fails to be re-created post build env expansion due to error: '%s'", url, e.toString()));
+                                continue;

```

Comment:
Seems like if expansion fails you can still do the comparison with the unexpanded value. Perhaps the catch clause should just log it, "urlWithExpandedParams = remoteURL" and let the remaining code execute.

Reply:
If the try part fails, likely urlWithExpandedParams will be null and the areSameGitRepo fails, but as it's just one line of difference, sure removed "continue;" to let the remaining code execute as suggested. Changed the logging part as well.